<a href="https://colab.research.google.com/github/jessicabolz/DATA-690-WANG/blob/master/Assignment_10extra.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Assignment # 10 - Point Estimate and Interval Estimate (Confidence Interval)

## A random survey of enrollment at 35 community colleges across the United States yielded the following figures:

6,414; 1,550; 2,109; 9,350; 21,828; 4,300; 5,944; 5,722; 2,825; 2,044;

5,481; 5,200; 5,853; 2,750; 10,012; 6,357; 27,000; 9,414; 7,681; 3,200;

17,500; 9,200; 7,380; 18,314; 6,557; 13,713; 17,768; 7,493; 2,771; 2,861;

1,263; 7,285; 28,165; 5,080; 11,622

## Perform point estimate and interval estimate with 95% confidence level using t-distribution.

## Since we don't know the population variance, we use t-distribiution instead of normal distribution.

In [38]:
# Import Python packages
import pandas as pd
import numpy as np
import statistics as st
import scipy.stats as stats


### Step 0 - Data Preprocessing

Process the raw data to make a list of integers. In order to calculate descriptive statistic, Python needs to work with a list of numbers.

In [6]:
# make each line of numbers a string object and then concatenate them together 
# The end result is one single string containing 35 numbers separated by ";"

data_1="6,414; 1,550; 2,109; 9,350; 21,828; 4,300; 5,944; 5,722; 2,825; 2,044;"
data_2="5,481; 5,200; 5,853; 2,750; 10,012; 6,357; 27,000; 9,414; 7,681; 3,200;"
data_3="17,500; 9,200; 7,380; 18,314; 6,557; 13,713; 17,768; 7,493; 2,771; 2,861;"
data_4="1,263; 7,285; 28,165; 5,080; 11,622"

data=data_1+data_2+data_3+data_4
print(data)

6,414; 1,550; 2,109; 9,350; 21,828; 4,300; 5,944; 5,722; 2,825; 2,044;5,481; 5,200; 5,853; 2,750; 10,012; 6,357; 27,000; 9,414; 7,681; 3,200;17,500; 9,200; 7,380; 18,314; 6,557; 13,713; 17,768; 7,493; 2,771; 2,861;1,263; 7,285; 28,165; 5,080; 11,622


In [7]:
type(data)

str

In [9]:
# Convert the single string to a list of strings using split() function
# Make sure to specify a delimter or separator

x=data.split(';')
print(x)

['6,414', ' 1,550', ' 2,109', ' 9,350', ' 21,828', ' 4,300', ' 5,944', ' 5,722', ' 2,825', ' 2,044', '5,481', ' 5,200', ' 5,853', ' 2,750', ' 10,012', ' 6,357', ' 27,000', ' 9,414', ' 7,681', ' 3,200', '17,500', ' 9,200', ' 7,380', ' 18,314', ' 6,557', ' 13,713', ' 17,768', ' 7,493', ' 2,771', ' 2,861', '1,263', ' 7,285', ' 28,165', ' 5,080', ' 11,622']


In [13]:
# Create a list of integers from the list of strings using List Comprehension or a function
# Make sure to remove the "," first and then convert the strings to integers

e=[]

for i in x:
  e.append(int(i.replace(",","")))
  

print(e)
type(e[0])


[6414, 1550, 2109, 9350, 21828, 4300, 5944, 5722, 2825, 2044, 5481, 5200, 5853, 2750, 10012, 6357, 27000, 9414, 7681, 3200, 17500, 9200, 7380, 18314, 6557, 13713, 17768, 7493, 2771, 2861, 1263, 7285, 28165, 5080, 11622]


int

### Step 1 - Calculate and Display the Sample Size and Sample Mean¶

In [24]:
# Calculate and display the sample size

print("Sample size: ",len(e))


Sample size:  35


In [53]:
# Calculate and display the sample mean

m=np.mean(e)
print("Sample mean: ",round(m,0))

Sample mean:  8629.0


The point estimate of the mean enrollment of US community colleges is 8629.

### Step 2 - Calculate and Display the Sample Standard Deviation & Sample Standard Error
* Sample Standard Deviation $S=\sqrt{\dfrac{1}{n-1}\sum\limits_{i=1}^n (X_i-\bar{X})^2}$

* Sample Standard Error = $\dfrac{S}{\sqrt{n}}$

Note: The default Delta Degree of Freedom (DDOF) for Numpy's std function is 0 which is applicable to populate data. For sample data, we need to specify ddof=1.

For the enrollment data, we round up the statistics to be the full integers (no decimal points).

In [45]:
# Calculate and display the sample standard deviation using Numpy's std function.

s=np.std(e,ddof=1)
print("Sample Standard Deviation:  ",round(s,0))

Sample Standard Deviation:   6944.0


In [62]:
# Calculate and display the sample standard error

stder=s/(35**0.5)
print("Sample Standard Error:  ",round(stder,0))

Sample Standard Error:   1174.0


### Step 3 - Calculate t Critical Value using t-Distribution
$\alpha$ = 1 - Confidence Level = 1 - 95% = 0.05

$\dfrac{\alpha}{2}$ = 0.025

n (sample size) = 35

df (degree of freedom) = n - 1 = 35 - 1 = 34

We will use Python scipy.stats t-distribution's PPF (Percentage Point Function) to calculate t critical value $t_{0.025,34}$.

In [47]:
# Calculate and display the t critical value using scipy.stats.t package ppf function

t_value = stats.t.ppf(0.975, 34)
print("t critical value or t statistics = ", round(t_value,2))



t critical value or t statistics =  2.03


### Step 4 - Calculate the Margin of Error
Margin of Error = t-Statistics * Sample Standard Error = $t_{\alpha/2,n-1}\left(\dfrac{s}{\sqrt{n}}\right)$

In [64]:
# Calculate and display the margin of error

merr=t_value*stder
print("Margin of error = ",round(merr,0))


Margin of error =  2385.0


### Step 5 - Calculate Lower and Upper Limit of the Confidence Interval

* Lower Limit = Sample Mean - Margin of Error
* Upper Limit = Sample Mean + Margin of Error

In [55]:
# Calculate and display the lower limit

lower_limit=m-merr
print("Lower limit = ",round(lower_limit,0))

Lower limit =  6243.0


In [56]:
# Calculate and display the upper limit

upper_limit=m+merr
print("Upper limit = ", round(upper_limit,0))

Upper limit =  11014.0


### Step 6 - Now We have the 95% Confidence Interval

*Confidence Interval ($\sigma$ unknown) = $\bar{x}\space\pm\space t_{\alpha/2}\left(\dfrac{S}{\sqrt{n}}\right)$ = Sample_Mean $\pm$ Margin of Error

In [60]:
# Print the 95% confidence interveral in the form (lower limit, upper limit)

print("The 95% Confidence Interval = ",round(lower_limit,0),round(upper_limit,0))

The 95% Confidence Interval =  6243.0 11014.0


### The End